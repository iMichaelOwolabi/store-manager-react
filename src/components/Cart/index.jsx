import React, { Fragment } from 'react';
import AuthNav from '../AuthNav';

const Cart = () => (
  <Fragment>
    <AuthNav />
    <div className="container">
      <h1>Cart</h1>
    </div>
  </Fragment>
);
export default Cart;
