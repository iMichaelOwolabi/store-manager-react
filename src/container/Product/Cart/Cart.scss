.cartHeader {
  margin-top: 4rem;
}
h2.topSpace {
  margin-bottom: 3rem;
}
input.qty {
  float: left;
  outline: none;
}
button.salesButton {
  padding: .8rem;
  border-radius: 1rem;
  font-weight: 800;
  font-size: 1.3rem;
  float: right;
}
table.responsiveTable {
  overflow-x: auto;
  margin: auto;
}