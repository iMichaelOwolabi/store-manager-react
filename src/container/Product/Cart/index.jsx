/* eslint-disable react/jsx-one-expression-per-line */
import React, { Fragment } from 'react';
import { Header, Table, Image } from 'semantic-ui-react';
import AuthNav from '../../../components/AuthNav';
import './Cart.scss';

const choiceProduct = JSON.parse(sessionStorage.getItem('cart'));

class Cart extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      products: {
        quantity: sessionStorage.getItem('qty')
      }
    };
  }

  handleChange = (event) => {
    const { products } = this.state;
    const { name, value } = event.target;
    products[name] = value;
    this.setState({
      products
    });
    sessionStorage.setItem('qty', products.quantity);
  }

  render() {
    const { products } = this.state;
    return (
      <Fragment>
        <AuthNav />
        <div className="wrapper">
          <div className="cartHeader" />
          <h2 className="topSpace">Your Shopping Cart</h2>
          <Table celled padded>
            <Table.Header>
              <Table.Row>
                <Table.HeaderCell>S/N</Table.HeaderCell>
                <Table.HeaderCell>Picture</Table.HeaderCell>
                <Table.HeaderCell>Quantity</Table.HeaderCell>
                <Table.HeaderCell>Price</Table.HeaderCell>
                <Table.HeaderCell>Total</Table.HeaderCell>
              </Table.Row>
            </Table.Header>

            <Table.Body>
              <Table.Row>
                <Table.Cell>
                  <Header as="h2">1</Header>
                </Table.Cell>
                <Table.Cell>
                  <Image src={choiceProduct.productimage} rounded size="mini" />
                </Table.Cell>
                <Table.Cell><input type="number" value={products.quantity} name="quantity" className="qty" onChange={this.handleChange} /></Table.Cell>
                <Table.Cell>&#8358; {choiceProduct.price}</Table.Cell>
                <Table.Cell>
                  &#8358;
                  {parseFloat(choiceProduct.price * products.quantity)}
                </Table.Cell>
              </Table.Row>
            </Table.Body>
          </Table>
          <button type="submit" className="salesButton">Make Sales</button>
        </div>
      </Fragment>
    );
  }
}

export default Cart;
