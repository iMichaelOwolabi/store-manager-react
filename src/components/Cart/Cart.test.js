import React from 'react';
import { shallow } from 'enzyme';
import Cart from '.';

describe('<Narbar Component />', () => {
  it('should', () => {
    const wrapper = shallow(<Cart />);
    expect(wrapper).toMatchSnapshot();
  });
});
