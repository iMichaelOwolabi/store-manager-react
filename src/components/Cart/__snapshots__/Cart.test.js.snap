// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Narbar Component /> should 1`] = `
<Fragment>
  <AuthNav />
  <div
    className="container"
  >
    <h1>
      Cart
    </h1>
  </div>
</Fragment>
`;
